tests/integration/components/spread-test.js

```diff
@@ -1,30 +1,235 @@
-/* jshint expr:true */
-import { expect } from 'chai'
-import {
-  describeComponent,
-  it
-} from 'ember-mocha'
+import Ember from 'ember'
+import {expect} from 'chai'
+import {$hook, initialize as initializeHook} from 'ember-hook'
+import {describeComponent, it} from 'ember-mocha'
 import hbs from 'htmlbars-inline-precompile'
+import {beforeEach, describe} from 'mocha'
+import sinon from 'sinon'
+import SpreadMixin from 'ember-spread'
 
-describeComponent(
-  'spread',
-  'Integration: SpreadComponent',
-  {
-    integration: true
-  },
-  function () {
-    it('renders', function () {
-      // Set any properties with this.set('myProperty', 'value');
-      // Handle any actions with this.on('myAction', function(val) { ... });
-      // Template block usage:
-      // this.render(hbs`
-      //   {{#spread}}
-      //     template content
-      //   {{/spread}}
-      // `);
-
-      this.render(hbs`{{spread}}`)
-      expect(this.$()).to.have.length(1)
-    })
-  }
-)
+import {integration} from 'dummy/tests/helpers/ember-test-utils/describe-component'
+
+describeComponent(...integration('spread'), function () {
+  let handler
+
+  beforeEach(function () {
+    initializeHook()
+
+    this.registry.register('component:spread-test', Ember.Component.extend(SpreadMixin, {
+
+      // == Properties ============================================================
+
+      hook: 'spread-test',
+      layout: hbs`
+        <div data-test={{hook 'spread-property'}}>
+          {{property}}
+        </div>
+      `,
+
+      // == Actions ===============================================================
+
+      click () {
+        this.onClick()
+      }
+    }))
+
+    handler = sinon.spy()
+  })
+
+  describe('when using the default (options) spread property', function () {
+    beforeEach(function () {
+      this.setProperties({
+        options: {
+          property: 'Neat',
+          onClick: handler
+        }
+      })
+
+      this.render(hbs`
+        {{spread-test
+          options=options
+        }}
+      `)
+    })
+
+    it('should bind spread properties as local properties', function () {
+      expect($hook('spread-property').text().trim()).to.equal('Neat')
+    })
+
+    it('should bind spread functions as local functions', function () {
+      $hook('spread-test').click()
+
+      expect(handler).to.have.callCount(1)
+    })
```

Comment:
A couple suggestions here. 

1) The reason this is a test for what it says it's a test for is b/c of the implementation of the `click` action in the `spread-test` component above, right? If it used `this.get('onClick')()` instead of `this.onClick()` then the description would no longer be valid, correct? Something about that bothers me, maybe we should just add a comment explaining that, because just looking at this test, it doesn't look like the `it` text is accurate, even though it is. 

2) I really try to put all actions into `beforeEach()` so you can describe what it is that you're doing separate from what it is you're expecting to happen after that action is performed. In this case, however, you're not actually testing that when the thing is clicked it calls the handler, you're confirming something more internal, the way in which the spread property is set on the component itself, so it probably wouldn't make as much sense to say something like: 

```js
describe('when spread-test is clicked', function () {
  beforeEach(function () {
    $hook('spread-test').click()
  })

  it('should call the locally bound function', function () {
    expect(handler).to.have.callCount(1)
  })
})
```